src/clouds/owncloud.js

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (C) 2018 Orange.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Rombit - initial API and implementation
+ ******************************************************************************/
+
+// const debug = require('debug-levels')('agile-data');
+// const agile = require('./agile-sdk');
+// const get = require('lodash/get');
+const path = require('path');
+const request = require('request');
+
+module.exports = {
+  endpointDescription: function () {
+    return({
+      "cloud": "owncloud",
+    })
+  },
+  upload: function(credentials, remotepath,payload) {
+    return new Promise(function(resolve, reject) {
+
+      if (typeof remotepath !== 'undefined' && typeof remotepath !== "undefined"  && typeof payload !== "undefined") {
+      let uploadConfig = {
+        uploadContent: JSON.stringify(payload),
+        uploadFileName: (typeof remotepath !== 'undefined') ? remotepath : dateFormat(new Date(), "yyyy.mm.dd") + "-Agile.txt",
+        uploadDirectory: (typeof remotepath !== 'undefined') ? path.dirname(remotepath) : '',
+        uploadDirectories: (typeof remotepath !== 'undefined') ? path.dirname(remotepath).split('/') : [],
+        loopCount: 0
```

Comment:
This value is never incremented, and hard coded to be 0, should the user have a way to change it?